beacon-chain/blockchain/service.go

```diff
@@ -15,26 +15,30 @@ var log = logrus.WithField("prefix", "blockchain")
 // ChainService represents a service that handles the internal
 // logic of managing the full PoS beacon chain.
 type ChainService struct {
-	ctx               context.Context
-	cancel            context.CancelFunc
-	beaconDB          *database.DB
-	chain             *BeaconChain
-	web3Service       *powchain.Web3Service
-	latestBeaconBlock chan *types.Block
-	processedHashes   [][32]byte
+	ctx                              context.Context
+	cancel                           context.CancelFunc
+	beaconDB                         *database.DB
+	chain                            *BeaconChain
+	web3Service                      *powchain.Web3Service
+	latestBeaconBlock                chan *types.Block
+	processedBlockHashes             [][32]byte
```

Comment:
Will these contain recently processed hashes, or every hash from genesis/last justified block? If the latter, will we load them all from disk at startup?

Reply:
these hash -> block/state mappings are just temporary until we have a trie structure (binary merkle or sparse markle?) to store that permanently in disk

Reply:
gotcha :+1: